cmd/create/idp/htpasswd.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright (c) 2022 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package idp
+
+import (
+	"fmt"
+	"os"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/ocm"
+)
+
+const ClusterAdminUsername = "cluster-admin"
+
+func createHTPasswdIDP(cmd *cobra.Command,
+	cluster *cmv1.Cluster,
+	clusterKey string,
+	idpName string,
+	ocmClient *ocm.Client) {
+	var err error
+	username := args.htpasswdUsername
+	password := args.htpasswdPassword
+
+	if username == "" || password == "" {
+		reporter.Infof("At least one user is required to create the IDP.")
+		username, password = getUserDetails(cmd, username, err, password)
+	}
+
+	// Choose which way to create the IDP according to whether it already has an admin or not.
+	htpasswdIDP := FindExistingHTPasswdIDP(cluster, ocmClient)
+	if htpasswdIDP != nil {
+		containsAdminOnly := !HasClusterAdmin(htpasswdIDP) || htpasswdIDP.Htpasswd().Users().Len() == 0
+		if !containsAdminOnly {
+			exitHTPasswdCreate("Cluster already has an HTPasswd IDP", nil)
+		}
+		// Existing IDP contains only admin. Add new user to it
+		reporter.Infof("Cluster already has an HTPasswd IDP, new user will be added to it")
+		err = ocmClient.AddHTPasswdUser(username, password, cluster.ID(), htpasswdIDP.ID())
+		if err != nil {
+			reporter.Errorf(
+				"Failed to add a user to the HTPasswd IDP of cluster '%s': %v", clusterKey, err)
+			os.Exit(1)
+		}
+		reporter.Infof("User '%s' added", username)
+	} else {
+		// HTPasswd IDP does not exist - create it
+		idpBuilder := cmv1.NewIdentityProvider().
+			Type("HTPasswdIdentityProvider").
+			Name(idpName).
+			Htpasswd(
+				cmv1.NewHTPasswdIdentityProvider().Users(
+					cmv1.NewHTPasswdUserList().Items(CreateHTPasswdUser(username, password)),
+				),
+			)
+		htpasswdIDP = doCreateIDP(idpName, *idpBuilder, cluster, clusterKey, ocmClient)
+	}
+
+	for shouldAddAnotherUser() {
```

Comment:
This should only be called if `interactive.Enabled()` otherwise it will prevent customers from using this in pipelines

Reply:
I didn't know that. But that would only allow them to add one user without the `--interactive` flag. I guess that's not too too bad for now because later on we intend to allow batch user addition.